### PR TITLE
[Service Connector] `az spring connection create`: Enable user-assigned managed identity for spring apps and deprecate Postgresql single server

### PR DESCRIPTION
**Pull request by wchigit**
**Related command**

az spring connection create [target-resource-type] --user-identity client-id=[client-id] subs-id=[subscription-id]

az webapp connection create postgres
az spring connection create postgres
az containerapp connection create postgres

**Description**

User-assigned managed identity is enabled as an auth type for spring apps to create connections.

Deprecate the Postgresql.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

**Review by yungezz — DISMISSED** [review]

File: src/azure-cli/azure/cli/command_modules/serviceconnector/commands.py [hunk 1]
```diff
@@ -55,15 +55,15 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
             for target in supported_target_resources:
                 with self.command_group('{} connection create'.format(source.value),
                                         connection_type, client_factory=cf_linker) as ig:
-                    if target == RESOURCE.Mysql:
+                    if target == RESOURCE.Mysql or target == RESOURCE.Postgres:
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/commands.py:58]: may define the targets to be deprecated in `_resource_configs.py`

File: src/azure-cli/azure/cli/command_modules/serviceconnector/tests/latest/recordings/test_springcloud_cosmossql_umi.yaml [hunk 2]
```diff
@@ -0,0 +1,936 @@
...
+      ParameterSetName:
+      - --id
+      User-Agent:
+      - AZURECLI/2.48.0 azsdk-python-mgmt-servicelinker/1.2.0b1 Python/3.9.7 (Windows-10-10.0.22621-SP0)
+    method: POST
+    uri: https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/servicelinker-test-linux-group/providers/Microsoft.AppPlatform/Spring/servicelinker-springcloud/apps/cosmossql/deployments/mock-deployment/providers/Microsoft.ServiceLinker/linkers/cosmos_2e20e/listConfigurations?api-version=2022-11-01-preview
+  response:
+    body:
+      string: '{"configurations": [{"name": "azure.cosmos.uri", "value": "https://servicelinker-sql-cosmos.documents.azure.com:443/"},
+        {"name": "azure.cosmos.key", "value": "4AfKEpKdsxibgloPh0Y4dXB8nSNzddl8HRzku3vFz9PpLF11JvYiuydgkZJNOasED5mGm8hIDfTX6XGyQR5PUQ=="},
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/tests/latest/recordings/test_springcloud_cosmossql_umi.yaml:420]: need hide these credentials, consider using/adding `recording_processors` in you tests

also check other cases in CredScan results

**Review by houk-ms — CHANGES_REQUESTED** [review]
**PR title**
Adding new auth types is a customer facing feature and we need use `[Service Connector]`

**CredScan**
Check current `recording_precessor` in the ScenarioTest def and adding your own rules to hide credentials

**Comment by houk-ms** [comment]
Not sure whether the title could be parsed or not, a better way is to create two PRs for two feature changes :)

**Review by xfz11 — APPROVED** [review]

**Review by houk-ms — APPROVED** [review]
hi @kairu-ms, please help review the PR when available

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**